File: fix_gi.py
import urllib.request
import urllib.parse
import json

url = "https://xbanoipgoleuahwbqksy.supabase.co/rest/v1/pedidos?responsavel=ilike.*Giovana*&select=id,responsavel"
headers = {
    "apikey": "sb_publishable_RSQ4odG0wxy8ZucJHu_WvQ_0JfM8jbh",
    "Authorization": "Bearer sb_publishable_RSQ4odG0wxy8ZucJHu_WvQ_0JfM8jbh",
    "Content-Type": "application/json",
    "Prefer": "return=representation"
}

req = urllib.request.Request(url, headers=headers)
try:
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode())
except Exception as e:
    print("Error fetching data:", e)
    exit(1)

to_update = []
for d in data:
    resp = d.get('responsavel', '')
    if not resp: continue
    
    parts = [s.strip() for s in resp.split(',')]
    new_parts = []
    has_duplicates = False
    
    for p in parts:
        if p not in new_parts:
            new_parts.append(p)
        else:
            has_duplicates = True
            
    if has_duplicates:
        to_update.append({
            'id': d['id'],
            'old': resp,
            'new': ', '.join(new_parts)
        })

print(f"Found {len(to_update)} rows to update.")

for u in to_update:
    patch_url = f"https://xbanoipgoleuahwbqksy.supabase.co/rest/v1/pedidos?id=eq.{u['id']}"
    payload = json.dumps({"responsavel": u['new']}).encode('utf-8')
    patch_req = urllib.request.Request(patch_url, data=payload, headers=headers, method='PATCH')
    try:
        with urllib.request.urlopen(patch_req) as response:
            print(f"- OS #{u['id']}: {u['old']} -> {u['new']}")
    except Exception as e:
        print(f"Error updating OS #{u['id']}:", e)

print("Finished updates.")

File: fix_gi.js
const { createClient } = require('@supabase/supabase-js');

const supabase = createClient(
    'https://xbanoipgoleuahwbqksy.supabase.co',
    'sb_publishable_RSQ4odG0wxy8ZucJHu_WvQ_0JfM8jbh'
);

async function main() {
    const { data, error } = await supabase.from('pedidos').select('id, responsavel').ilike('responsavel', '%Gi%');
    if (error) {
        console.error("Error:", error);
        return;
    }
    
    // We only want exact matches of 'Gi', 'gi', 'Gi,', ', Gi', ', Gi,', etc.
    const toUpdate = [];
    for (let d of data) {
        if (!d.responsavel) continue;
        let arr = d.responsavel.split(',').map(s => s.trim());
        let hasGi = false;
        let newArr = [];
        for (let r of arr) {
            if (r.toLowerCase() === 'gi') {
                hasGi = true;
                // Wait, should we remove it entirely or replace with Giovana?
                // The user said "preciso que seja removi". Let's replace it with 'Giovana' just in case, but if it already has 'Giovana', we don't need to add it again. Or I can just remove it, but then the O.S. will have no responsible if it was only Gi. I will replace it with 'Giovana'.
                if (!newArr.includes('Giovana')) {
                    newArr.push('Giovana');
                }
            } else {
                newArr.push(r);
            }
        }
        
        if (hasGi) {
            toUpdate.push({
                id: d.id,
                old: d.responsavel,
                new: newArr.join(', ')
            });
        }
    }
    
    console.log(`Found ${toUpdate.length} rows to update.`);
    for (let u of toUpdate) {
        console.log(`- OS #${u.id}: ${u.old} -> ${u.new}`);
    }
    
    // Actually update
    for (let u of toUpdate) {
        await supabase.from('pedidos').update({ responsavel: u.new }).eq('id', u.id);
    }
    console.log("Updates applied successfully.");
}

main();
